proto/substrait/extensions/extensions.proto

```diff
@@ -64,8 +64,29 @@ message SimpleExtensionDeclaration {
 
     // A simple name if there is only one impl for the function within the YAML.
     // A compound name, referencing that includes type short names if there is
-    // more than one impl per name in the YAML.
+    // more than one impl per name in the YAML. A simple name not found in the
+    // YAML if "udf" is given.
     string name = 3;
+
+    UserDefinedFunction udf = 4;
+
+    message UserDefinedFunction {
+      // A serialization of the user-defined function's code. The default serialization
+      // format is using base64 of cloudpickle.dumps().
+      string code = 1;
```

Comment:
Why use a UTF-8 `string` and then use base64 when you can just use a protobuf [`binary`](https://developers.google.com/protocol-buffers/docs/proto3#scalar) (bottom of the table)? Generally, I feel like this should be a protobuf [`Any`](https://developers.google.com/protocol-buffers/docs/proto3#any), which pretty much just is a standardized format for a `binary` and a `string` that acts as a format specification.

Reply:
The reason is no longer relevant, so I'll fix this and then `base64` would not be needed. I think the type can be `bytes`. The format of the bytes would be `cloudpickle` by default.

Reply:
Note that if we go with the [above-described alternative](https://github.com/substrait-io/substrait/pull/234#issuecomment-1167301202) then `cloudpickle` format would be default only in the `PythonPickleFunction` case.